src/main/java/com/google/sps/util/DataStoreHelper.java

```diff
@@ -30,14 +31,17 @@ public static Date parseInputDate(String textDate){
     }
     
     public static long queryUserID(String username) throws com.google.cloud.datastore.DatastoreException {
-        String gqlQuery = USER_QUERY_TEMPLATE + username;
-
+        Query<Entity> query =
+          Query.newEntityQueryBuilder()
+            .setKind("User")
+            .setFilter(PropertyFilter.eq("username", username))
+            .build();
         Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
-        Query<?> query = Query.newGqlQueryBuilder(gqlQuery).build();
-        QueryResults<?> results = datastore.run(query);
-        
-        long userID = ((BaseEntity<Key>) results.next()).getLong("userId");
-        
+        QueryResults<Entity> results = datastore.run(query);
+        if(!results.hasNext())
+            throw new DatastoreException(123,"not found", "for same name");
```

Comment:
Could you explain the parameters?

Reply:
it just need this three parameters, but we doesn't actaully need them to deliver any info, what we actaully need is just to catch the exception, the parameters doesn't actaully matters. Thus I create them randomly.